canvas_modules/common-canvas/src/object-model/api-pipeline.js

```diff
@@ -101,9 +101,10 @@ export default class APIPipeline {
 		return linksToDelete;
 	}
 
-	isEmpty() {
+	isEmpty(includeLinks) {
 		if (this.getNodes() && this.getNodes().length === 0 &&
-				this.getComments() && this.getComments().length === 0) {
+				this.getComments() && this.getComments().length === 0 &&
+				this.getLinks() && this.getLinks().length === 0) {
```

Comment:
Should `getLinks()` be called only if `includeLinks` is true?

Reply:
Hi Carita - good catch. Actually I introduced the `includeLinks` param because I initially thought that this check needs to be dependent on whether detachable links is switched on or not. However, I realized that if link selection is switched off there cannot be any detached links without there being at last a couple of nodes/comments. So I removed the `includeLinks` param from the code but forgot to remove it from the function signature. (If you trace back through the code it is not actually passed in from here in common-canvas.jsx:
```js
isCanvasEmpty={this.canvasController.isPrimaryPipelineEmpty()}
```